cmd/shield/main.go

```diff
@@ -1817,15 +1830,27 @@ func main() {
 				return err
 			}
 
-			if err := RestoreArchive(id, string(b)); err != nil {
+			var taskUUID string = ""
+			if taskUUID, err = RestoreArchive(id, string(b)); err != nil {
 				return err
 			}
 
 			targetMsg := ""
 			if params.Target != "" {
 				targetMsg = fmt.Sprintf("to target '%s'", params.Target)
 			}
-			OK("Scheduled immediate restore of archive '%s' %s", id, targetMsg)
+			if taskUUID != "" && *opts.Raw {
+				RawJSON(map[string]interface{}{
+					"ok":        fmt.Sprintf("Scheduled immediate restore of archive '%s' %s", id, targetMsg),
+					"task_uuid": taskUUID,
+				})
+			} else {
+				OK("Scheduled immediate restore of archive '%s' %s", id, targetMsg)
+				if taskUUID != "" {
+					ansi.Printf("To view task, type @B{shield show task %s}\n", taskUUID)
+				}
+			}
+
```

Comment:
as above, it _looks_ like the if/else could lead to a case where `--raw` returns non-JSON